### ISSUE TITLE
Create components

### ISSUE THREAD

**Issue by NathanWalker** [NathanWalker/nativescript-angular-web-components#2]
https://docs.nativescript.org/ui/ui-views
- [x] Button
- [x] Label
- [ ] TextField
- [ ] TextView
- [ ] SearchBar
- [ ] Switch
- [ ] Slider
- [ ] Progress
- [ ] ActivityIndicator
- [x] Image
- [x] ListView
- [ ] HtmlView
- [ ] WebView
- [ ] TabView
- [ ] SegmentedBar
- [ ] DatePicker
- [ ] TimePicker
- [ ] ListPicker
- [ ] Dialogs

**Comment by OscarAgreda** [comment]
Why don't you just use the existing angular material design 2 instead of creating new web components. So if you have a TabView on NativeScript just tell angular 2 to create a md-tab on the web. And we all know the power of the material design directives.

**Comment by jrood** [comment]
@OscarAgreda are you talking about [angular/material2](https://github.com/angular/material2)? That's a cool idea, but I think that project is still very much in alpha.

**Comment by OscarAgreda** [comment]
@jrood 
yes i am referring to angular/material2.
I think there are 2 options 

1- Convert/Transpose/translate each NativeScript component into their matching angular/material2 directive.
or
2- Create the web components from scratch to match each NativeScript Ui view., this also means "CSS maintenance" which will become time consuming.

i use angular-material version 1 in production and is so well done and awesome . I don't think the material design team will take too much longer to release a good version of Angular Material Design 2 , considering that Angular 2 is almost out. Therefore with that assumption , I think option 2 makes more sense.

**Comment by jrood** [comment]
@OscarAgreda good point. I hadn't thought about css maintenance.

**Comment by prijindal** [comment]
What about using ionic 2 for this
They guys have already made components which work with angular 2 and those guys are also using typescript which follow more or less the standards of material design

**Comment by NathanWalker** [comment]
@prijindal That's a really great idea :)
Ionic 2 is fantastic, love it. It works wonderfully for web and hybrid apps.
In this case, it could render Ionic web components to match the NativeScript equivalents.
Duly noted, this will be attempted in the future. Thanks 👍

**Comment by prijindal** [comment]
If you guys need any help, I can help with making and integrating these components.
And I am sure that other people from Ionic community will be excited about something like this

**Comment by NathanWalker** [comment]
@prijindal we would love your help.
A good place to start would be just with the `ActionBar` to use the `ion-toolbar`:
http://ionicframework.com/docs/v2/components/#toolbar

If you could help integrate Ionic 2 and get that working would be awesome.

**Comment by prijindal** [comment]
I will look into it

**Comment by sean-perkins** [comment]
@NathanWalker Are you looking for the web equivalents to the {N} components? If so, I have a few of them already baked; just would need to abstract out my app branding logic from them. 

i.e. SegmentedBar for Web Declaration

`<sd-segmentedbar [items]="segmentedItems" (selectedIndexChanged)="selectedIndexChanged($event)"></sd-segmentedbar>`

**Comment by sipacate** [comment]
Are people still interested in the functionality offered by this repo?

**Comment by lukashlobil** [comment]
I learned to live without it, would still love to see the components though

**Comment by jrood** [comment]
Yes definitely. In fact, OscarAgreda's suggestion to use the Angular 2 Material components seems like an even better idea to me now that I've used some of those components.

**Comment by maddes** [comment]
I can't think of a library I want more to become reality. I'll try helping in the future.
About the material2 subject, I think it's still to much in alpha. Ionic seems more suitable for now. Maybe we can use a common base but instead of naming the platform "web" we name it "ionic" or "angular" or "vue" so we can eventually move to the one that proves best.
